src/pages/workspace/workflows/approvals/WorkspaceWorkflowsApprovalsApproverPage.tsx

```diff
@@ -1,45 +1,249 @@
 import useNetwork from '@hooks/useNetwork';
+import useThemeStyles from '@hooks/useThemeStyles';
+import * as DeviceCapabilities from '@libs/DeviceCapabilities';
 import {formatPhoneNumber} from '@libs/LocalePhoneNumber';
 import Log from '@libs/Log';
 import Navigation from '@libs/Navigation/Navigation';
 import type {FullScreenNavigatorParamList} from '@libs/Navigation/types';
 import * as OptionsListUtils from '@libs/OptionsListUtils';
+import Permissions from '@libs/Permissions';
 import * as PersonalDetailsUtils from '@libs/PersonalDetailsUtils';
 import * as PolicyUtils from '@libs/PolicyUtils';
 import AccessOrNotFoundWrapper from '@pages/workspace/AccessOrNotFoundWrapper';
 import withPolicyAndFullscreenLoading from '@pages/workspace/withPolicyAndFullscreenLoading';
 import type {WithPolicyAndFullscreenLoadingProps} from '@pages/workspace/withPolicyAndFullscreenLoading';
 import * as Policy from '@userActions/Policy/Policy';
+import * as Workflow from '@userActions/Workflow';
 import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
 import type SCREENS from '@src/SCREENS';
-import type {PersonalDetailsList, PolicyEmployee} from '@src/types/onyx';
+import type {Beta, PolicyEmployee} from '@src/types/onyx';
+import type {Icon} from '@src/types/onyx/OnyxCommon';
 import {isEmptyObject} from '@src/types/utils/EmptyObject';
 
 type WorkspaceWorkflowsApprovalsApproverPageOnyxProps = {
-    /** All of the personal details for everyone */
-    personalDetails: OnyxEntry<PersonalDetailsList>;
+    /** Beta features list */
+    // eslint-disable-next-line react/no-unused-prop-types -- This prop is used in the component
+    betas: OnyxEntry<Beta[]>;
 };
 
 type WorkspaceWorkflowsApprovalsApproverPageProps = WorkspaceWorkflowsApprovalsApproverPageOnyxProps &
     WithPolicyAndFullscreenLoadingProps &
     StackScreenProps<FullScreenNavigatorParamList, typeof SCREENS.WORKSPACE.WORKFLOWS_APPROVALS_APPROVER>;
+
+type SelectionListApprover = {
+    text: string;
+    alternateText: string;
+    keyForList: string;
+    isSelected: boolean;
+    login: string;
+    rightElement?: React.ReactNode;
+    icons: Icon[];
+};
+type ApproverSection = SectionListData<SelectionListApprover, Section<SelectionListApprover>>;
+
+function WorkspaceWorkflowsApprovalsApproverPage(props: WorkspaceWorkflowsApprovalsApproverPageProps) {
+    if (Permissions.canUseWorkflowsAdvancedApproval(props.betas) && props.route.params.approverIndex !== undefined) {
+        // eslint-disable-next-line react/jsx-props-no-spreading
+        return <WorkspaceWorkflowsApprovalsApproverPageNew {...props} />;
+    }
+
+    // eslint-disable-next-line react/jsx-props-no-spreading
+    return <WorkspaceWorkflowsApprovalsApproverPageOld {...props} />;
+}
+
+function WorkspaceWorkflowsApprovalsApproverPageNew({policy, personalDetails, isLoadingReportData = true, route}: WorkspaceWorkflowsApprovalsApproverPageProps) {
```

Comment:
Would you mind changing the name of this component to `WorkspaceWorkflowsApprovalsApproverPageBeta` and I would just keep the name of the "old" component the way it was before. The only reason is that "new" and "old" hold no meaning once this PR is merged.